### ISSUE TITLE
Metamask Saving Data From Network Form

### ISSUE THREAD

**Issue by mjosephd** [MetaMask/metamask-extension#11010]
I do realize this may not be a issue technically, but I opened a support ticket without any response and couldn't find any other channel for communication/ Thus posting here.

So basically I am trying to do some changes on the metamask extension, add a new field to the Network Settings, a checkbox specifically.

So I managed to add this checkbox on the screen (use the appropriate storybook component, etc). However, I am now struggling to figure out how metamask persists its data permanently and loads them.

Basically I added a new field, that field is visible to the user, it also gets sent as part of the `UPDATE_METAMASK_STATE`. It is a sibling field of `blockExplorerUrl`, so I sort of modeled it after that. However the value does not seem to get saved to the store that persists the data. So my question is

1. How do I persist data in general and when a new field is added to the `network-form.component.js`
2. How is data persisted and loaded into metamask (data like settings, list of custom networks)
